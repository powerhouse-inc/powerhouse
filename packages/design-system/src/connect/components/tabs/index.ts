export * from "./tab-content.js";
export * from "./tabs.js";
