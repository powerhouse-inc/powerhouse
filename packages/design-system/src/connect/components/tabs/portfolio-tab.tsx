import { AssetsTable, TabContent } from "@/rwa";

export function PortfolioTab() {
  return (
    <TabContent
      description="Details on the distribution of assets among different financial institutions or investment vehicles."
      label="Portfolio"
    >
      <AssetsTable />
    </TabContent>
  );
}
