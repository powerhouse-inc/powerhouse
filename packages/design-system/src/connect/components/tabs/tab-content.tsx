import { type ReactNode } from "react";

export type TabContentProps = {
  readonly label: ReactNode;
  readonly description: ReactNode;
  readonly children: ReactNode;
  readonly disabled?: boolean;
};
export function TabContent(props: TabContentProps) {
  const { label, children } = props;

  return <div>{children}</div>;
}
