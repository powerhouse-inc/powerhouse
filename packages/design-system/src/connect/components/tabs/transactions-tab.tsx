import { GroupTransactionsTable, TabContent } from "@/rwa";

export function TransactionsTab() {
  return (
    <TabContent
      description="Details on the distribution of assets among different financial institutions or investment vehicles."
      label="Transactions"
    >
      <GroupTransactionsTable />
    </TabContent>
  );
}
