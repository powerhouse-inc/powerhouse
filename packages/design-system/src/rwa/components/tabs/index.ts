export * from "./other-tab";
export * from "./portfolio-tab";
export * from "./tab-content";
export * from "./tabs";
export * from "./transactions-tab";
