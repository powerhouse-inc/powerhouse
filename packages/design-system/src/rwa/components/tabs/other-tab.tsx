import { tableNames, TableWithForm } from "@/rwa";
import { AccountsTable } from "../table/accounts";

export function OtherTab() {
  return (
    <div className="grid grid-rows-4 gap-4">
      <div>
        <h2 className="mb-2 text-lg font-bold">Accounts</h2>
        <p className="mb-4 text-xs text-gray-600">
          Add and manage accounts that can be associated with transactions and
          fees.
        </p>
        <AccountsTable />
      </div>
      <div>
        <h2 className="mb-2 text-lg font-bold">Fixed Income Types</h2>
        <p className="mb-4 text-xs text-gray-600">
          Add and manage fixed income asset types.
        </p>
        <TableWithForm tableName={tableNames.FIXED_INCOME_TYPE} />
      </div>
      <div>
        <h2 className="mb-2 text-lg font-bold">SPVs</h2>
        <p className="mb-4 text-xs text-gray-600">
          Add and manage special purpose vehicles (SPVs).
        </p>
        <TableWithForm tableName={tableNames.SPV} />
      </div>
      <div>
        <h2 className="mb-2 text-lg font-bold">Service Provider Fee Types</h2>
        <p className="mb-4 text-xs text-gray-600">
          Add and manage service providers and their associated fee types.
        </p>
        <TableWithForm tableName={tableNames.SERVICE_PROVIDER_FEE_TYPE} />
      </div>
    </div>
  );
}
