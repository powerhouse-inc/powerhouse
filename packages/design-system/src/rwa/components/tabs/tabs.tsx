import { EditorActionButtons } from "@/connect";
import {
  OtherTab,
  PortfolioTab,
  TransactionsTab,
  useEditorContext,
} from "@/rwa";
import { Content, List, Root, Trigger } from "@radix-ui/react-tabs";
import { useCallback } from "react";

export function RWATabs() {
  const tabs = ["Portfolio", "Transactions", "Other"] as const;
  const {
    onSwitchboardLinkClick,
    onExport,
    onClose,
    onShowRevisionHistory,
    clearSelected,
  } = useEditorContext();

  const handleTabChange = useCallback(() => {
    clearSelected();
  }, [clearSelected]);

  return (
    <Root defaultValue="Portfolio" onValueChange={handleTabChange}>
      <div className="flex justify-between">
        {/* <EditorUndoRedoButtons {...props} /> */}
        <List className="flex gap-x-2 rounded-xl bg-slate-50 p-1 text-sm font-semibold text-gray-600 outline-none">
          {tabs.map((tab) => (
            <Trigger
              className="data-[state='active']:tab-shadow data-disabled:cursor-not-allowed data-disabled:text-gray-400 h-7 w-32 rounded-lg  transition duration-300 data-[state='active']:bg-gray-50 data-[state='active']:text-gray-900"
              key={tab}
              value={tab}
            >
              {tab}
            </Trigger>
          ))}
        </List>
        <EditorActionButtons
          onClose={onClose}
          onExport={onExport}
          onShowRevisionHistory={onShowRevisionHistory}
          onSwitchboardLinkClick={onSwitchboardLinkClick}
        />
      </div>
      <div className="mt-3 rounded-md bg-slate-50 p-8">
        <Content value="Portfolio">
          <PortfolioTab />
        </Content>
        <Content value="Transactions">
          <TransactionsTab />
        </Content>
        <Content value="Other">
          <OtherTab />
        </Content>
      </div>
    </Root>
  );
}
