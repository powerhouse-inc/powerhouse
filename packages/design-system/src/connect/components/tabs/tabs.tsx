import { Content, List, Root, Trigger } from "@radix-ui/react-tabs";
import React from "react";
import { type TabContentProps } from "./tab-content.js";

export function Tabs({
  children,
  defaultValue,
}: {
  children: React.ReactNode;
  defaultValue: string;
}) {
  return (
    <Root defaultValue={defaultValue} className="flex flex-col gap-2">
      <div className="flex w-full justify-between">
        {/* <EditorUndoRedoButtons {...props} /> */}
        <List className="flex w-full gap-x-2 rounded-xl bg-slate-50 p-1 text-sm font-semibold text-gray-600 outline-none">
          {React.Children.map(children, (child, i) => {
            if (!React.isValidElement(child)) return;
            const { label, disabled } = child.props as TabContentProps;
            return (
              <Trigger
                className="data-[state='active']:tab-shadow ata-disabled:cursor-not-allowed data-disabled:text-gray-400 flex h-7 flex-1 items-center justify-center rounded-lg transition duration-300 data-[state='active']:bg-gray-50 data-[state='active']:text-gray-900"
                key={label as string}
                value={label as string}
                disabled={disabled ?? false}
              >
                {label as string}
              </Trigger>
            );
          })}
        </List>
      </div>
      <div className="mt-3 rounded-md bg-white">
        {React.Children.map(children, (child, i) => {
          if (!React.isValidElement(child)) return;
          const { label } = child.props as TabContentProps;
          return (
            <Content value={label as string} key={i}>
              {child}
            </Content>
          );
        })}
      </div>
    </Root>
  );
}
