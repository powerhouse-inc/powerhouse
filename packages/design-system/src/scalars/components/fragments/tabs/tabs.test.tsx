import { render, screen } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import { Tabs, TabsContent, TabsList, TabsTrigger } from "./tabs.js";

describe("Tabs Components", () => {
  it("should  Tabs match snapshot", () => {
    const { asFragment } = render(<Tabs />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should  TabsList match snapshot", () => {
    const { asFragment } = render(
      <Tabs>
        <TabsList />
      </Tabs>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("should TabsContent  match snapshot", () => {
    const { asFragment } = render(
      <Tabs>
        <TabsContent value="tab1">Content 1</TabsContent>
      </Tabs>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("should TabsTrigger match snapshot", () => {
    const { asFragment } = render(
      <Tabs>
        <TabsList>
          <TabsTrigger value="tab1">Tab 1</TabsTrigger>
        </TabsList>
      </Tabs>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders basic tabs structure", () => {
    render(
      <Tabs defaultValue="tab1">
        <TabsList>
          <TabsTrigger value="tab1">Tab 1</TabsTrigger>
          <TabsTrigger value="tab2">Tab 2</TabsTrigger>
        </TabsList>
        <TabsContent value="tab1">Content 1</TabsContent>
        <TabsContent value="tab2">Content 2</TabsContent>
      </Tabs>,
    );
    expect(screen.getByRole("tablist")).toBeInTheDocument();
    expect(screen.getAllByRole("tab")).toHaveLength(2);
    expect(screen.getByText("Content 1")).toBeInTheDocument();
  });

  it("activates when selected", async () => {
    render(
      <Tabs defaultValue="tab1">
        <TabsList>
          <TabsTrigger value="tab1">Tab 1</TabsTrigger>
          <TabsTrigger value="tab2">Tab 2</TabsTrigger>
        </TabsList>
      </Tabs>,
    );
    const firstTab = screen.getAllByRole("tab")[0];
    expect(firstTab).toHaveAttribute("data-state", "active");
    await userEvent.click(screen.getByText("Tab 2"));
    expect(screen.getByText("Tab 2")).toHaveAttribute("data-state", "active");
  });

  it("shows content when associated tab is active", async () => {
    render(
      <Tabs defaultValue="tab1">
        <TabsList>
          <TabsTrigger value="tab1">Tab 1</TabsTrigger>
          <TabsTrigger value="tab2">Tab 2</TabsTrigger>
        </TabsList>
        <TabsContent value="tab1" data-testid="content-1">
          Content 1
        </TabsContent>
        <TabsContent value="tab2" data-testid="content-2">
          Content 2
        </TabsContent>
      </Tabs>,
    );

    // Check initial state
    expect(screen.getByTestId("content-1")).toHaveAttribute(
      "data-state",
      "active",
    );
    expect(screen.getByTestId("content-2")).toHaveAttribute(
      "data-state",
      "inactive",
    );

    // Switch tabs
    await userEvent.click(screen.getByText("Tab 2"));

    // Verify updated state
    expect(screen.getByTestId("content-1")).toHaveAttribute(
      "data-state",
      "inactive",
    );
    expect(screen.getByTestId("content-2")).toHaveAttribute(
      "data-state",
      "active",
    );
  });
});
