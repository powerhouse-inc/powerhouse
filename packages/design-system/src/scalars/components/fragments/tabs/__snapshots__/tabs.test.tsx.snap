// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tabs Components > should  Tabs match snapshot 1`] = `
<DocumentFragment>
  <div
    data-orientation="horizontal"
    dir="ltr"
  />
</DocumentFragment>
`;

exports[`Tabs Components > should  TabsList match snapshot 1`] = `
<DocumentFragment>
  <div
    data-orientation="horizontal"
    dir="ltr"
  >
    <div
      aria-orientation="horizontal"
      class="bg-muted text-muted-foreground inline-flex h-9 items-center justify-center rounded-lg p-1"
      data-orientation="horizontal"
      role="tablist"
      style="outline-color: none; outline-style: none; outline-width: initial;"
      tabindex="-1"
    />
  </div>
</DocumentFragment>
`;

exports[`Tabs Components > should TabsContent  match snapshot 1`] = `
<DocumentFragment>
  <div
    data-orientation="horizontal"
    dir="ltr"
  >
    <div
      aria-labelledby="radix-:r2:-trigger-tab1"
      class="ring-offset-background focus-visible:ring-ring mt-2 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2"
      data-orientation="horizontal"
      data-state="inactive"
      hidden=""
      id="radix-:r2:-content-tab1"
      role="tabpanel"
      tabindex="0"
    />
  </div>
</DocumentFragment>
`;

exports[`Tabs Components > should TabsTrigger match snapshot 1`] = `
<DocumentFragment>
  <div
    data-orientation="horizontal"
    dir="ltr"
  >
    <div
      aria-orientation="horizontal"
      class="bg-muted text-muted-foreground inline-flex h-9 items-center justify-center rounded-lg p-1"
      data-orientation="horizontal"
      role="tablist"
      style="outline-color: none; outline-style: none; outline-width: initial;"
      tabindex="0"
    >
      <button
        aria-controls="radix-:r3:-content-tab1"
        aria-selected="false"
        class="ring-offset-background focus-visible:ring-ring data-[state=active]:bg-background data-[state=active]:text-foreground inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1 text-sm font-medium transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="inactive"
        id="radix-:r3:-trigger-tab1"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Tab 1
      </button>
    </div>
  </div>
</DocumentFragment>
`;
